components/script/dom/document.rs

```diff
@@ -2010,8 +2006,7 @@ impl Document {
         // TODO
         // Step 8.2. For each deferred fetch record deferredRecord of navigable’s active document’s
         // relevant settings object’s fetch group’s deferred fetch records:
-        for deferred_fetch in control_document.loader().deferred_fetches() {
-            let deferred_fetch = deferred_fetch.lock().unwrap();
+        for deferred_fetch in navigable.as_global_scope().deferred_fetches() {
```

Comment:
Why not make this a method of fetch group, that way you can operate on references instead?